src/v/pandaproxy/server.cc

```diff
@@ -88,6 +92,17 @@ struct handler_adaptor : ss::httpd::handler_base {
         server::reply_t rp{std::move(rep)};
         auto req_size = get_request_size(*rq.req);
         auto sem_units = co_await ss::get_units(_ctx.mem_sem, req_size);
+
+        ss::sstring access_log;
+        if (_log.is_enabled(ss::log_level::debug)) {
+            access_log = ssx::sformat(
+              R"("{} {} HTTP/{}")",
+              rq.req->_method,
+              rq.req->_url,
+              rq.req->_version);
+            vlog(_log.debug, "{}", access_log);
```

Comment:
Oh, whoops. That was the intention; I added this `vlog` here as debug code. The log below uses `access_log`, so it's all on one line.

Reply:
Two drawbacks of one line per request:
* we don't know when the call had started (neither time nor log position)
* if the process `abort`s in the handler (like on a bad_alloc) we will have lost the context in the log

Since it's DEBUG level, maybe we can log method/url/version twice? Like this:
```
DEBUG 2022-11-01 18:00:05,202 [shard 0] rest_proxy - server.cc:128 - "GET /topics HTTP/1.1" {
...
DEBUG 2022-11-01 18:00:05,305 [shard 0] rest_proxy - server.cc:128 - "GET /topics HTTP/1.1" } 200 2
```

Here the `{` and `}` would indicate entering/leaving the handler

Reply:
No objection to having two lines, the key thing is that there should be one line that logs the request along with its result status -- having an additional line to mark the start of the request is okay too.

I think it makes sense for the start+finish to be at different severities, e.g. in this case start at TRACE and finish at DEBUG, so that user can get the "normal" request logging (i.e. similar to what they would get with nginx or apache) without both lines.

Reply:
I'm not a fan of the `{` `}`, to me it implies a scope, but there could be very many unrelated lines between them.

Reply:
Anything that is brief, intuitve, and distict enough from the rest of the line (to be easily grep-able) would do instead of curly braces, some ideas:
* `<` `>` ("message in/out")
* `@in` `@out`
* `(E)`nter `(L)`eave

Reply:
@BenPope what does HA Proxy or the other well known proxies do? there may be a standard format we can borrow from them, so log parsers already work w/out inventing our own custom format/parser.

Reply:
also, what about having one line w/ all the info. i.e.: a `finally` style log. 

```
initial_ctx = ...{};
...work()
...then()
.then()
.finally([ ]{ log (initial_ctx, final_ctx)}
```

Reply:
I think we should stick to one line per access and it should be formulated roughly as you suggested, which is below: https://github.com/redpanda-data/redpanda/pull/7049/commits/3b0ca7e3983743822bf3b3d46032cbc30f0f552d#diff-c00e17022dc1dc55b0fb2cb1e028642a172b7a50b58fd127867f57863c165f2aR121-R128

The format is slightly different to say the [Common Log Format](https://en.wikipedia.org/wiki/Common_Log_Format) because I didn't want to make our logs incompatible with our logs, which have a common prefix (outisde of backtraces and another anomaly) e.g.:

This PR:
```
level date shard service - source_location - request status bytres
```
Common Log Format:
```
host ident authuser date request status bytes
```
I think it makes sense to add `host`, `ident` if they're handy.